### DISCUSSION TITLE
Pentair IntelliFlo VSF Standalone via Elfin EW11A no RX Packets / TX Packets Fail

### DISCUSSION THREAD

**Discussion by EPiSKiNG** [tagyoureit/nodejs-poolController#878] [RS485] [answered]
Greetings!

I have a Pentair IntelliFlo VSF and no additional Pentair products at the moment (no IntelliCenter or anything like that).

I also have an Elfin EW11 that is wired up to the pump with the Yellow wire connected to A+ and Green to B-  I am seeing the following:

![Screenshot 2024-01-08 095518](https://github.com/tagyoureit/nodejs-poolController/assets/737924/c31747e8-ea18-4057-896c-d65a7686da92)
![Screenshot 2024-01-08 095734](https://github.com/tagyoureit/nodejs-poolController/assets/737924/ec5a8ebc-c7a6-4e84-b431-78acede258bb)
![image](https://github.com/tagyoureit/nodejs-poolController/assets/737924/e57b955c-9b00-4bdf-ad6f-64fcde1a6d11)
![Screenshot 2024-01-08 095811](https://github.com/tagyoureit/nodejs-poolController/assets/737924/df00ccf0-53ea-48f4-b902-d1d13b9b2c47)
![Screenshot 2024-01-08 095846](https://github.com/tagyoureit/nodejs-poolController/assets/737924/94fa2cb0-788d-4b36-ad57-6ee54bf79c15)

I'm perplexed at why I'm not receiving anything back to Nixie / njsPC... I can confirm that pings to the device are functioning properly between the two nodes.

**Comment by rstrouse** [comment]
Is this an intelliflo3?

**Reply by EPiSKiNG** [reply]
I'm not sure... Doesn't say IntelliFlo3 anywhere... Heres a photo:
![20240106_163447](https://github.com/tagyoureit/nodejs-poolController/assets/737924/a1da8260-8fb0-4783-8cfb-66a3c3d02299)
![20240106_163450](https://github.com/tagyoureit/nodejs-poolController/assets/737924/0e120248-1b31-47da-8287-62d1d59471d0)
![20240109_071658](https://github.com/tagyoureit/nodejs-poolController/assets/737924/acaf96ff-57ab-4cf8-a863-24e2e312924a)

**Reply by rstrouse** [reply]
This is not an IntelliFlo 3 it is one of the older models and should work just fine.

First add the pool circuit to the configuration.  Lets have something to turn on and off.  If I were to interpret the the screenshots above it looks like Nixie is sending packets and the adapter is sending them to the pump.  But the pump is not responding.

This typically means one or more of a few things.  Either the address currently configured on the pump does not match or this is one of the models that has other settings on the pump to allow it to be controlled by RS485 or there is a wiring issue.  
1. Please check the manual to ensure the pump settings are set up for RS485 automation
2. Please check the manual to ensure the pump is on address 1 or change Nixie to the address that is assigned in the pump.
3. Double check the wiring.  Make sure the connections are good and perhaps swap the green/yellow wires.  This would not be the first RS485 adapter that had the printing for the connectors wrong.

**Reply by EPiSKiNG** [reply]
Pump is assigned address 1.
Checked the manual for the pump and I'm only seeing the following information:
![Untitled](https://github.com/tagyoureit/nodejs-poolController/assets/737924/3c20036b-4253-44be-b70e-2dfd0280123d)

I've tried swapping green/yellow (A+/B-) wires several times and neither way works.

I'm going to rebuild Nixie/njsPC from scratch on a brand new Ubuntu 22.04 VM to see if there may have been some sort of issue I caused when setting it up.  Also going to try some debugging from my windows host with a virtual COM port to see if I'm getting any data back from the Elfin.

**Comment by EPiSKiNG** [comment]
Here is a log from a fresh install of njsPC:

I did not install REM, and I am using NP2, with the section about REM commented out... I couldn't get REM to install, and I really don't think I need it since I'm just trying to control a single pump via RS485 over TCP (I am running njsPC on a virtual machine, not a pi)


> 1|njsPC      | [1/10/2024, 2:44:34 AM] info: [2:44:34 AM] 10.100.42.87 GET /config/options/pumps?null {}
> 1|njsPC      | [1/10/2024, 2:44:34 AM] info: [2:44:34 AM] 10.100.42.87 PUT /config/pump {"id":50,"master":1,"name":"IntellifloVSF","type":4,"portId":0,"address":96,"model":null,"primingTime":0,"primingSpeed":0,"minSpeed":450,"maxSpeed":3450,"minFlow":15,"maxFlow":130,"circuits":[]}
> 1|njsPC      | [1/10/2024, 2:44:40 AM] info: [2:44:40 AM] 10.100.42.87 GET /config/options/general?null {}
> 1|njsPC      | [1/10/2024, 2:44:47 AM] info: [2:44:47 AM] 10.100.42.87 PUT /app/rs485Port {"enabled":true,"netConnect":true,"mock":false,"rs485Port":"/dev/ttyUSB0","netPort":8899,"netHost":"192.168.200.235","portId":0,"portSettings":{"baudRate":9600,"dataBits":8,"parity":"none","stopBits":1},"netSettings":{"allowHalfOpen":false,"keepAlive":false,"keepAliveInitialDelay":5},"type":"netConnect","inactivityRetry":10}
> 1|njsPC      | [1/10/2024, 2:44:47 AM] error: Unable to close the current RS485 port
> 1|njsPC      | [1/10/2024, 2:44:52 AM] info: [2:44:52 AM] 10.100.42.87 PUT /app/rs485Port {"enabled":true,"netConnect":true,"mock":false,"rs485Port":"/dev/ttyUSB0","netPort":8899,"netHost":"192.168.200.235","portId":0,"portSettings":{"baudRate":9600,"dataBits":8,"parity":"none","stopBits":1},"netSettings":{"allowHalfOpen":false,"keepAlive":false,"keepAliveInitialDelay":5},"type":"netConnect","inactivityRetry":10}
> 1|njsPC      | [1/10/2024, 2:44:52 AM] info: Net connect (socat) 0 connected to: 192.168.200.235:8899
> 1|njsPC      | [1/10/2024, 2:44:52 AM] info: Net connect (socat) Connection 0 connected
> 1|njsPC      | [1/10/2024, 2:44:52 AM] info: Net connect (socat) 0 ready and communicating: 192.168.200.235:8899
> 1|njsPC      | sendScreenlogicStats set to true
> 1|njsPC      | sendScreenlogicStats set to true
> 1|njsPC      | [1/10/2024, 2:44:56 AM] warn: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:44:56 AM] error: Error sending requestPumpStatus for IntellifloVSF: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:44:58 AM] warn: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:44:58 AM] error: Error sending setPumpToRemoteControl for IntellifloVSF: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:45:01 AM] warn: dt:Wed Jan 10 2024 02:40:04 GMT+0000 (Coordinated Universal Time) lat:undefined lon:undefined Not enough information to calculate Heliotrope.  See https://github.com/tagyoureit/nodejs-poolController/issues/245
> 1|njsPC      | [1/10/2024, 2:45:01 AM] warn: dt:Wed Jan 10 2024 02:40:04 GMT+0000 (Coordinated Universal Time) lat:undefined lon:undefined Not enough information to calculate Heliotrope.  See https://github.com/tagyoureit/nodejs-poolController/issues/245
> 1|njsPC      | [1/10/2024, 2:45:01 AM] warn: dt:Wed Jan 10 2024 02:40:04 GMT+0000 (Coordinated Universal Time) lat:undefined lon:undefined Not enough information to calculate Heliotrope.  See https://github.com/tagyoureit/nodejs-poolController/issues/245
> 1|njsPC      | [1/10/2024, 2:45:01 AM] warn: dt:Wed Jan 10 2024 02:40:04 GMT+0000 (Coordinated Universal Time) lat:undefined lon:undefined Not enough information to calculate Heliotrope.  See https://github.com/tagyoureit/nodejs-poolController/issues/245
> 1|njsPC      | [1/10/2024, 2:45:01 AM] warn: dt:Wed Jan 10 2024 02:40:04 GMT+0000 (Coordinated Universal Time) lat:undefined lon:undefined Not enough information to calculate Heliotrope.  See https://github.com/tagyoureit/nodejs-poolController/issues/245
> 1|njsPC      | [1/10/2024, 2:45:01 AM] warn: dt:Wed Jan 10 2024 02:40:04 GMT+0000 (Coordinated Universal Time) lat:undefined lon:undefined Not enough information to calculate Heliotrope.  See https://github.com/tagyoureit/nodejs-poolController/issues/245
> 1|njsPC      | [1/10/2024, 2:45:02 AM] warn: Message aborted after 2 attempt(s): 165,0,96,33,6,1,4,1,49
> 1|njsPC      | [1/10/2024, 2:45:02 AM] error: Error sending setDriveState for IntellifloVSF: Message aborted after 2 attempt(s): 165,0,96,33,6,1,4,1,49
> 1|njsPC      | [1/10/2024, 2:45:06 AM] warn: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:45:06 AM] error: Error sending requestPumpStatus for IntellifloVSF: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:45:08 AM] warn: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:45:08 AM] error: Error sending setPumpToRemoteControl for IntellifloVSF: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:45:12 AM] warn: Message aborted after 2 attempt(s): 165,0,96,33,6,1,4,1,49
> 1|njsPC      | [1/10/2024, 2:45:12 AM] error: Error sending setDriveState for IntellifloVSF: Message aborted after 2 attempt(s): 165,0,96,33,6,1,4,1,49
> 1|njsPC      | [1/10/2024, 2:45:16 AM] warn: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:45:16 AM] error: Error sending requestPumpStatus for IntellifloVSF: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:45:18 AM] warn: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:45:18 AM] error: Error sending setPumpToRemoteControl for IntellifloVSF: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:45:22 AM] warn: Message aborted after 2 attempt(s): 165,0,96,33,6,1,4,1,49
> 1|njsPC      | [1/10/2024, 2:45:22 AM] error: Error sending setDriveState for IntellifloVSF: Message aborted after 2 attempt(s): 165,0,96,33,6,1,4,1,49
> 1|njsPC      | [1/10/2024, 2:45:26 AM] warn: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:45:26 AM] error: Error sending requestPumpStatus for IntellifloVSF: Message aborted after 3 attempt(s): 165,0,96,33,7,0,1,45
> 1|njsPC      | [1/10/2024, 2:45:28 AM] warn: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:45:28 AM] error: Error sending setPumpToRemoteControl for IntellifloVSF: Message aborted after 2 attempt(s): 165,0,96,33,4,1,255,2,42
> 1|njsPC      | [1/10/2024, 2:45:28 AM] info: [2:45:28 AM] 10.100.42.87 PUT /app/rs485Port {"enabled":false,"netConnect":true,"mock":false,"rs485Port":"/dev/ttyUSB0","netPort":8899,"netHost":"192.168.200.235","portId":0,"portSettings":{"baudRate":9600,"dataBits":8,"parity":"none","stopBits":1},"netSettings":{"allowHalfOpen":false,"keepAlive":false,"keepAliveInitialDelay":5},"type":"netConnect","inactivityRetry":10}
> 1|njsPC      | [1/10/2024, 2:45:28 AM] info: Net connect (socat) 0 request close: 192.168.200.235:8899
> 1|njsPC      | [1/10/2024, 2:45:28 AM] info: Net connect (socat) 0 sent FIN packet: 192.168.200.235:8899
> 1|njsPC      | [1/10/2024, 2:45:28 AM] info: Net connect (socat) destroy socket: 192.168.200.235:8899
> 1|njsPC      | [1/10/2024, 2:45:28 AM] info: Net connect (socat) 0 successfully closed: 192.168.200.235:8899
> 1|njsPC      | sendScreenlogicStats set to true
> 1|njsPC      | sendScreenlogicStats set to true


I also used Realterm to connect to the Elfin EW11 directly from my Windows PC, and I am able to connect, but obviously not seeing any traffic since I haven't sent any "hello" to "wake up" the RS485 link to the Intelliflo... Is there some command I can send via a direct RS485 link to kick my pump into sending back data?

As it is, I'm not getting anything back from the pump.  As far as wires being backwards, my understanding is that I would still see traffic, but that it would be "inverted"?

I'm stumped... Might give up on this Elfin RS485 over TCP device; bite the bullet and a pi + hat. :(

**Comment by EPiSKiNG** [comment]
Well, I've tried 2 additional USB RS485 adapters, a raspberry pi4 and a laptop, all with no avail... I did at some point get the pump into "Display Not Active" mode, and got 1 successful received packet, and I have no idea how it happened, and I can't seem to get it to replicate.  I've re-terminated the cables no less than 5 times, and even tried a completely different Pentair comm cable.

Still have no idea why I can't get any successful sent packets... I can't imagine the wiring is THAT sensitive is it?

**Comment by EPiSKiNG** [comment, marked as answer]
Well, I feel like an idiot. :)

The rs485 connector inside the control panel area of the pump was loose.  Reseated it, and it's working like a charm!

**Comment by mguinness** [comment]
Glad that you got it working.  For reference a photo of the connector below taken from [RS-485 Connection](https://www.troublefreepool.com/wiki/index.php?title=Pentair_IntelliFlo_Pumps#RS-485_Connection).

![Connector](https://www.troublefreepool.com/wiki/images/c/c3/Pentair_Intelliflo_Loose_RS-485_Connection.jpg)

FWIW based on your photos I think you have a [IntelliFlo VSF - 011056](https://www.pentair.com/en-us/products/residential/pool-spa-equipment/pool-pumps/intelliflo_vsf/sku/011056.html).  Keep us posted on your progress.

**Comment by EPiSKiNG** [comment]
Yep!  That's where the problem was! 

Now I'm trying to determine two things:

How do I set up a schedule so that the pump runs at 40gpm from 730am to 815am, then 20gpm from 815am to 11am?

How do I allow my pool guy (yeah I know... I'll take the responsibility over at some point!) To start an egg timer for 4 hours when he comes to add chemicals and perform back flush?  

I'm integrated into home assistant using the HACS plugin, so I'm thinking maybe some sort of outdoor button that can trigger the pump to run an egg timer schedule?  Or maybe I disable njsPC during the time schedule that the guy will be on site?  

Thanks!!!

**Comment by EPiSKiNG** [comment]
I figured out the schedules... Had to create circuits first, then configured them in the pump circuits, then the schedules.  Super!

Still figuring out how to handle the switch for the pool guy, but thinking a weatherproof box with a zigbee button in it.

**Comment by mguinness** [comment]
Regarding the pool guy keep it simple.  I think the Zigbee button would only confuse and using the pump keypad would work better.   I would mirror the schedule on the pump (40gpm from 730am to 815am, then 20gpm from 815am to 11am) so that you can disconnect njsPC during the pool service window and then the pool guy can then use the Quick Clean button (or program button) on the pump directly (which they should already be familiar with).  This also has the benefit that if njsPC fails while you are not home then the pump will continue to run during the scheduled times.

**Reply by rstrouse** [reply]
You can also trigger service mode from REM using a dry contact input.  This can be set to toggle the timeout function but this may not be the best because the timeout can restart as soon as the timer runs out.  However, then again your pool guy should not be disassembling equipment when the breakers are still on.

**Reply by EPiSKiNG** [reply]
I like the dry contact idea... Saves on battery costs too. :) Downside is I don't get to keep graphs of pump usage during that timeframe inside of Home Assistant.

Is REM or njsPC able to connect to a wifi-based relay or does it need to be physically connected to the njsPC instance?

**Reply by rstrouse** [reply]
REM and njsPC both have a rest interface as well as MQTT.  For REM you configure this as an event but it doesn't do polling on a Wifi location.  Why not trigger it from a GPIO at the pad?  If you l look through the discussions somebody did a scenario where they had two buttons where one set a timeout and the other turned off the timeout.  They also showed an LED indicating the panel was in timeout mode.
